### PR TITLE
Expose `set_compliance_policy` and `get_ciphers`

### PR DESCRIPTION
**Pull request by jhoyla**
Expose APIs to access `SSL_CTX_set_compliance_policy` and `SSL_CTX_get_ciphers`.

### CODE REVIEW

File: boring/src/ssl/mod.rs [hunk 1]
```diff
@@ -701,6 +701,27 @@ impl SslCurve {
     pub const P256_KYBER768_DRAFT00: SslCurve = SslCurve(ffi::NID_P256Kyber768Draft00);
 }
 
+/// A compliance policy
```
Comment (ghedo) [inline boring/src/ssl/mod.rs:704]:
```suggestion
/// A compliance policy.
```

File: boring/src/ssl/mod.rs [hunk 2]
```diff
@@ -1281,6 +1304,21 @@ impl SslContextBuilder {
...
 
+    /// Gets the list of supported ciphers for protocols before TLSv1.3.
+    ///
+    /// See [`ciphers`] for details on the format
+    ///
+    /// This corresponds to [`SSL_CTX_get_ciphers`].
+    ///
+    /// [`ciphers`]: https://www.openssl.org/docs/manmaster/man1/ciphers.html
+    /// [`SSL_CTX_set_cipher_list`]: https://www.openssl.org/docs/manmaster/man3/SSL_CTX_get_ciphers.html
+    pub fn get_ciphers(&mut self) -> Result<&StackRef<SslCipher>, ErrorStack> {
```
Comment (nox) [inline boring/src/ssl/mod.rs:1315]: This should take `&self` and you may want to put the same method on `SslContext` too.

Comment (nox) [inline boring/src/ssl/mod.rs:1315]: I don't think the Result is necessary here

https://github.com/google/boringssl/blob/b628f8721e7e67302b0e26c92e0108a31066194e/ssl/ssl_lib.cc#L2075-L2077

Seems like NULL should be interpreted as None, or even maybe that the method should panic on NULL.

https://github.com/google/boringssl/blob/master/ssl/internal.h#L631

**Review by nox — APPROVED** [review]

File: boring/src/ssl/mod.rs [hunk 3]
```diff
@@ -1281,6 +1304,25 @@ impl SslContextBuilder {
...
 
+    /// Gets the list of supported ciphers for protocols before TLSv1.3.
+    ///
+    /// See [`ciphers`] for details on the format
+    ///
+    /// This corresponds to [`SSL_CTX_get_ciphers`].
+    ///
+    /// [`ciphers`]: https://www.openssl.org/docs/manmaster/man1/ciphers.html
+    /// [`SSL_CTX_set_cipher_list`]: https://www.openssl.org/docs/manmaster/man3/SSL_CTX_get_ciphers.html
+    pub fn ciphers(&self) -> Option<&StackRef<SslCipher>> {
```
Comment (nox) [inline boring/src/ssl/mod.rs:1315]: `SslContextBuilder` owns a `SslContext`, so this should just be `self.0.ciphers()`

**Review by nox — CHANGES_REQUESTED** [review]

**Review by nox — APPROVED** [review]

**Pull request merged**